protocol.md

```diff
@@ -0,0 +1,196 @@
+
+However, Whisper was not designed to handle huge number of messages and real-time communication. These are the tradeoffs that we accepted when implementing this protocol.
+
+This protocol can either work using a Whisper service which requires a protocol implementation to run in the same process as a Whisper node as well as using a Whisper client which might run as a separate Ethereum node and communicate through IPC or WebSocket.
+
+There is some tight coupling between the payload and Whisper:
+* Whisper message topic depends on the actual message type
+* Whisper message uses a different key (asymmetric or symmetric) depending on the actual message type
+
+## Whisper node configuration
+
+If you want to run a Whisper node and receive messages from Status clients, you need to configure it properly, otherwise, some message might not be received.
+
+Whisper's Proof Of Work algorithm is used to to deter denial of service and various spam/flood attacks against the Whisper network. The sender of a message must perform some work which in this case means performing some computations. Because Status main client is a mobile client, this easily leads to battery draining and poor performance of the app itself. Hence, these clients send envelopes with the following settings:
+* proof-of-work equal `0.002`
+* proof-of-work time equal to `10`
+* time-to-live equal `10`
+
+If you want to receive messages from a mobile client, you need to run Whisper node with PoW set to `0.002`. In case of `geth`, this option can be overriden with `-shh.pow=0.002` flag.
+
+## Keys management
+
+TBD
+
+## Topic
+
+There are two types of Whisper topics the protocol uses:
+* static topic for private chats
+* dynamic topic based on a chat name for public chats
+
+Static topic is always the same and its hex representation is `0xf8946aac`. It is also called a discovery topic. Having only one topic for all private chats has an advantage as it's very hard to figure out who talks to who. A drawback is that everyone receives everyone's messages but they can decrypt only these they have private keys for.
+
+A dynamic topic is derived from a string using the following algorithm:
+
+```
+var hash []byte
+
+hash = keccak256(name)
+
+# Whisper V6 specific
+var topic [4]byte
+
+topic_len = 4
+
+if len(hash) < topic_len {
+    topic_len = len(hash)
+}
+
+for i = 0; i < topic_len; i++ {
+    topic[i] = hash[i]
+}
+```
+
+## Encryption
+
+Protocol distinguishes messages encrypted using asymmetric and symmetric encryptions.
+
+Symmetric keys are created using `shh_generateSymKeyFromPassword` Whisper JSON-RPC method which accepts one param, a string.
+
+Messages encrypted with asymmetric encryption should be encrypted using recipeint's public key so that only the recipient can decrypt them.
+
+# Perfect Forward Secrecy (PFS)
```

Comment:
Can fold in this https://github.com/status-im/status.im/blob/556217b63e77a38e8a7bfe9f2b748deacadd4fec/source/research/pfs.md